pkg/api/vlabs/const.go

```diff
@@ -47,6 +47,10 @@ const (
 	MinDiskSizeGB = 1
 	// MaxDiskSizeGB specifies the maximum attached disk size
 	MaxDiskSizeGB = 1024
+	// MinIPAddressCount specifies the minimum number of IP addresses per network interface
+	MinIPAddressCount = 1
+	// MaxIPAddressCount specifies the maximum number of IP addresses per network interface
+	MaxIPAddressCount = 256
```

Comment:
You were right that there is a limit, but it is 256. It turns out NRP enforces 256 total IP addresses. I tried creating an agent pool with 256, it worked.